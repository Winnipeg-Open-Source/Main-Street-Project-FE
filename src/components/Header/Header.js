import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex } from 'pcln-design-system';
import { Menu } from 'pcln-icons';

const StyledHeader = styled(Flex)`
    position: sticky;
    overflow-y: hidden;
    top: 0;
    align-items: center;
    user-select: none;
    font-weight: bold;
    box-shadow: 1px 2px 4px 0 rgba(0,0,0,0.08);
`;

const StyledMenu = styled(Menu)`
    cursor: pointer;
`;

function Header ({ className, onMenuClick }) {
    return (
        <StyledHeader className={className} color='primary' p={3}>
            <StyledMenu title='Menu' mr={3} onClick={onMenuClick} data-testid='menu-icon' />
            Main Street Project Inventory System
        </StyledHeader>
    );
}

Header.displayName = 'Header';

Header.propTypes = {
    className: PropTypes.string,
    onMenuClick: PropTypes.func,
};

Header.defaultProps = {
    className: '',
};

export default memo(Header);
