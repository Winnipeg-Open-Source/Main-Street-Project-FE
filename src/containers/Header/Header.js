import React from 'react';
import HeaderComponent from 'components/Header';
import useAuthentication from 'hooks/context/useAuthentication';
import useSidebar from 'hooks/context/useSidebar';

function Header () {
    const { isLoggedIn } = useAuthentication();
    const { onSidebarToggle } = useSidebar();

    return (
        <HeaderComponent isLoggedIn={isLoggedIn} onMenuClick={onSidebarToggle} />
    );
}

Header.displayName = 'Header';

export default Header;
