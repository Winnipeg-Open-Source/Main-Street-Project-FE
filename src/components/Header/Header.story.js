import React from 'react';
import { action } from '@storybook/addon-actions';
import Header from './Header';

export const Default = () => <Header onMenuClick={action('Menu Clicked')} />;

export default {
    title: 'Header',
    component: Header,
};
