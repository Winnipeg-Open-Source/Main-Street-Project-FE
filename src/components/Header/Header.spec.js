import React from 'react';
import { fireEvent } from '@testing-library/react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import Header from './Header';

describe ('Header', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<Header />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('displays correct name', () => {
        const { getByText } = renderWithTheme(<Header />);

        expect(getByText(/Main Street Project Inventory System/)).toBeInTheDocument();
    });

    it ('allows menu click', () => {
        const onMenuClick = jest.fn();
        const { getByTestId } = renderWithTheme(<Header onMenuClick={onMenuClick} />);

        const MenuIcon = getByTestId(/menu-icon/);

        expect(onMenuClick).toHaveBeenCalledTimes(0);
        fireEvent.click(MenuIcon);
        expect(onMenuClick).toHaveBeenCalledTimes(1);
    });
});
