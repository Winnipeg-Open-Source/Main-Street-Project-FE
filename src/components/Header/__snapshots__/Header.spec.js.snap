// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pbJYR hNzbLI sc-pkfPC fQMEfn"
  >
    <div
      class="sc-AxjAm sc-AxiKw sc-psDhf ksKCEt"
      color="primary"
    >
      <svg
        aria-hidden="true"
        class="sc-Axmtr dFmDHs sc-pAZqv sc-qQxFs hbnjDB"
        data-testid="menu-icon"
        fill="currentcolor"
        focusable="false"
        height="24"
        role="img"
        tabindex="-1"
        title="Menu"
        viewBox="0 0 24 24"
        width="24"
      >
        <title>
          Menu
        </title>
        <path
          d="M3 13h18v-2H3v2zm0 5h18v-2H3v2zM3 6v2h18V6H3z"
        />
      </svg>
      Main Street Project Inventory System
    </div>
  </div>
</DocumentFragment>
`;
